client/checkout/utils/upe.js

```diff
@@ -83,9 +58,11 @@ export const getHiddenBillingFields = ( enabledBillingFields ) => {
 	};
 };
 
-export const getUpeSettings = () => {
+export const getUpeSettings = ( paymentMethodType ) => {
```

Comment:
I decided to add `paymentMethodType` here because the current idea behind `getUpeSettings` is that we run it for every gateway/payment method and it should theoretically check for every payment method type. So we let `shouldIncludeTerms` to scope by the `wcpay-upe-form` which has the payment method type equal to the one coming as a parameter here, and check from there